File: easy/Road-Trip/Road-Trip.java
import java.io.*;
import java.util.Arrays;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
        ShowDistances(line);
            // Process line of input Here
        }
    }
    
    
    static void ShowDistances(String line) {
        String[] cities = line.split(";");
        int leng = cities.length;
        int[] distances = new int[leng];
        for (int i = 0; i < leng; i++) {
            int pos = cities[i].indexOf(",");
            distances[i] = Integer.valueOf(cities[i].substring(pos + 1));
        }
        Arrays.sort(distances);
        int i = 0;
        int last = 0;
        do {
            System.out.print(distances[i] - last);
            last = distances[i];
            i++;
            if (i < leng) {
                System.out.print(",");
            }
        } while (i < leng);
        System.out.println();

    }
}

File: easy/Road-Trip/Road-Trip.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowDistances(line);
        }
    }
    
    static void ShowDistances(string line) {
            
        string[] cities = line.Substring(0,line.Length-1).Split(';');
        int leng = cities.Length;
        int[] distances = new int[leng];
        for (int i = 0; i < leng; i++) {
            int pos = cities[i].IndexOf(',');
            distances[i] = System.Convert.ToInt32(cities[i].Substring(pos + 1));
        }
        Array.Sort(distances);
        int j = 0;
        int last = 0;
        do {
            System.Console.Write(distances[j] - last);
            last = distances[j];
            j++;
            if (j < leng) {
                System.Console.Write(",");
            }
        } while (j < leng);
        System.Console.WriteLine();

    }
}
